WireDomain/Sources/WireDomain/Repositories/Conversations/ConversationLocalStore.swift

```
@@ -0,0 +1,399 @@
+/// A local store dedicated to conversations.
+/// The store uses the injected context to perform `CoreData` operations
+/// on conversations objects.
+public protocol ConversationLocalStoreProtocol {
+
+    /// Stores the specified conversation
+    /// - Parameter conversation: The conversation to store.
+    func storeConversation(
+        _ conversation: WireAPI.Conversation,
+        isFederationEnabled: Bool
+    ) async throws
+
+    /// Deletes a local conversation given a qualified id.
+    /// - Parameter qualifiedId: The conversation qualified ID.
+
+    func deleteConversation(
+        withQualifiedId qualifiedId: WireAPI.QualifiedID
+    ) async throws
+
+    /// Removes a SelfUser from the specified conversation.
+    /// - Parameter qualifiedId: The conversation qualified ID.
+
+    func removeSelfUserFromConversation(
+        withQualifiedId qualifiedId: WireAPI.QualifiedID
+    ) async
+
+    /// Stores a flag indicating whether a conversation requires update from backend.
+    /// - Parameter qualifiedId: The conversation qualified ID.
+
+    func storeConversationNeedsBackendUpdate(
+        _ needsUpdate: Bool,
+        qualifiedId: WireAPI.QualifiedID
+    ) async
+
+    /// Stores the specified failed conversation
+    /// - Parameter qualifiedId: The conversation qualified ID.
+
+    func storeFailedConversation(
+        withQualifiedId qualifiedId: WireAPI.QualifiedID
+    ) async
+}
+
+public final class ConversationLocalStore: ConversationLocalStoreProtocol {
+
+    // MARK: - Properties
+
+    let context: NSManagedObjectContext
+
+    // MARK: - Object lifecycle
+
+    public init(
+        context: NSManagedObjectContext
+    ) {
+        self.context = context
+    }
+
+    // MARK: - Public
+
+    public func storeConversation(
+        _ conversation: WireAPI.Conversation,
+        isFederationEnabled: Bool
+    ) async throws {
+        guard let id = conversation.id ?? conversation.qualifiedID?.uuid else {
```

Is a missing ID a valid state? If no we should do something other than failing silently. Consider, throwing or logging the error and asserting. Also consider doing something in the default case below.

---

I will check that with backend team, thx

---

Actually, we are using the WireAPI.Conversation which is a new object that was recently created as part of the refactoring plan (see ticket https://wearezeta.atlassian.net/browse/WPB-8696 that is now closed) so we can assume that here we are consuming an object that is reliable.

---

Based on legacy codebase, we're not doing anything here:

```
        guard let conversationType = payload.type.map(BackendConversationType.clientConversationType) else {
            return nil
        }
```

and 

```
        default:
            return nil
```

We just proceed to the next conversation iteration. 
I can add some print statement for now, please note that this ticket is just about slow sync and we will most definitely work again on these objects later on which will enable us to make some adjustments if needed.